src/utils/mod.rs

```diff
@@ -616,6 +617,28 @@ pub(crate) mod txn_helpers {
                 "Fee input and fee change records' public key do not match".to_string(),
             ));
         }
+        // Check that the merkle path is valid w.r.t. ro, path and root.
+        // Only turned on in debugging mode.
+        #[cfg(debug_assertions)]
```

Comment:
Why only do this check in debug mode? I'd consider it to be an error if there are _any_ inputs to `generate` which yield `WrongQuotientPolyDegree` -- all of that should be checked before running the proof, and should always give better error messages than that.

Reply:
I guess @chancharles92 's concern was that the cost of this check will be non-negligible.
Also I believe the idea is that the caller should have checked the validity of `fee_info` before calling `generate` to build the proof.

Agree that error propagation should have been handled better than `WrongQuotientPolyDegree`.

Reply:
> Hmm, that's true, if the input is incorrect, it's always more efficient to exit early and output the correct error message. Maybe we can remove the debug mode? @zhenfeizhang

alright. Now it is removed.